board/templates/layout/pagination.html

```diff
@@ -2,25 +2,35 @@
         <ul class="pagination">
         {% if page_obj.has_previous %}
             <li class="page-item">
-                <a class="page-link" href="?page={{ page_obj.previous_page_number }}" aria-label="Previous">
+                <button class="page-link" onclick="myFunction({{ page_obj.previous_page_number }})" aria-label="Previous">
                     <span aria-hidden="true">&laquo;</span>
-                </a>
+                </button>
             </li>
         {% endif %}
             {% for item in page_numbers %}
                 {% if item == page_obj.number %}
                 <li class="page-item"><a class="page-link active">{{ item }}</a></li>
                 {% else %}
-                <li class="page-item"><a class="page-link" href="?page={{ item }}">{{ item }}</a></li>
+                <li class="page-item"><button class="page-link" onclick="myFunction({{ item }})">{{ item }}</button></li>
                 {% endif %}
             {% endfor %}
         {% if page_obj.has_next %}
             <li class="page-item">
-                <a class="page-link" href="?page={{ page_obj.next_page_number }}" aria-label="Next">
+                <button class="page-link" onclick="myFunction({{ page_obj.next_page_number }})" aria-label="Next">
```

Comment:
You have all the data in at render time, so you can create a [custom template filter](https://docs.djangoproject.com/en/3.2/howto/custom-template-tags/#writing-custom-template-filters) for this and do smgth like this without any javascript:
`href="{{ next_page_number|get_page_url }}`